src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const arrayResult = [];
+  let result = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      result = Object.assign(result, action.extraData);
```

Comment:
[DONT REPEAT YOURSELF] - Since you are creating a new copy of the state in each iteration at line 29, you don't need to assign the result back to `result` here. You can directly modify the `result` and then make a copy of it before pushing to `arrayResult`.